### PR TITLE
PS_v1

### PR DESCRIPTION
**Pull request by Rocky640**
A small speed-up on my workstation.
100% equivalent to base master.

Results for 5 tests for each version:

Base      Test      Diff
Mean    1441475   1447974   -6499
StDev   16288     15493     1893

p-value: 1
speedup: 0.005

### CODE REVIEW

**Comment by lucasart** [comment]
Speed up confirmed (i7-3770k, linux, gcc 4.9.1):

    stat        test     master    diff
    mean   2,459,081  2,446,694  12,387
    stdev     12,815     13,173   3,298
    
    speedup       0.51%
    P(speedup>0) 100.0%

File: src/pawns.cpp [hunk 1]
```diff
@@ -137,15 +137,14 @@ namespace {
...
-        // unsupported or connected (but not the backward one).
-        connected   =   ourPawns   & adjacent_files_bb(f) & (rank_bb(s) | p);
-        phalanx     =   connected  & rank_bb(s);
-        unsupported = !(ourPawns   & adjacent_files_bb(f) & p);
-        isolated    = !(ourPawns   & adjacent_files_bb(f));
+        // Flag the pawn
+        phalanx     = friends & rank_bb(s);
+        supported   = friends & rank_bb(s - Up);
+        //connected   = phalanx | supported;
+        //isolated    = !friends
```
Comment (glinscott) [inline src/pawns.cpp]: We can remove these commented out lines

**Comment by glinscott** [comment]
Nice!  Looks great.

**Comment by mcostalba** [comment]
I got a functional change here......

File: src/pawns.cpp [hunk 2]
```diff
@@ -155,8 +154,8 @@ namespace {
         // If the pawn is passed, isolated, connected or a lever it cannot be
         // backward. If there are friendly pawns behind on adjacent files
         // it cannot be backward either.
-        if (   (passed | isolated | lever | connected)
-            || (ourPawns & pawn_attack_span(Them, s)))
+        if (   (passed | !friends | lever | phalanx)
```
Comment (lucasart) [inline src/pawns.cpp]: why are we using `|` instead of `||` here? is this hack a mesurable speedup?

File: src/pawns.cpp [hunk 3]
```diff
@@ -193,7 +191,7 @@ namespace {
         if (backward)
             score -= Backward[opposed][f];
 
-        if (connected)
+        if (phalanx | supported)
```
Comment (lucasart) [inline src/pawns.cpp]: this is wrong. it should be `||` not `|`. `phalanx` is a bitboard, not a bool!

**Comment by mcostalba** [comment]
I took the freedom to push something similar, but less invasive, to C++11 branch. Only obvious and clean changes.

**Comment by Rocky640** [comment]
@lucasart: using || would introduce more branching. My educated guess is that | is faster,
but not equipped to measure this.
 
In the original code, if (   (passed | isolated | lever | connected) are all bools except the last one
connected (a bitboard) was moved to the end to avoid a compilation warning.

In my proposition, if (passed | !friends | lever | phalanx) 
friends is a bitboard
so the following would be better to avoid some compilation warning
if (passed | lever | phalanx | !friends)

if (phalanx | supported)
and 
if (phalanx || supported) 
should be equivalent, but I agree that the later is cleaner.

@mcostalba: I will make more tests on my side, because I did not see the functional change.

I came with those changes trying to make sense of what was the exact relationship between connected,unsupported and phalanx. 
I also felt that the code was more readable this way, but this is a matter of taste.


I will close the request for now, because all this might no longer be relevant once we migrate to c++11
See also the Apex patch, which just passed the LTC test which will need the supported bitboard.

**Pull request closed without merging**